### PR TITLE
Update Rust nightly version + Expose virtual-function-elimination

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

By enabling virtual-function-elimination we can reduce the size of the
final binaries. Here is a before and after for OpenTitan:

Before:
    Binary: 139K
    .text: 22818
After
    Binary: 136K
    .text: 21d0c

As virtual-function-elimination is still experimental let's not enable
it by default, but leave it as an option. This allows people to test it
and help contribute improvements and bug reports back to the Rust
compiler.

### Testing Strategy

CI

### TODO or Help Wanted

N/A

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

File: boards/Makefile.common [hunk 1]
```diff
@@ -266,6 +266,16 @@ else
   SHA256SUM := sha256sum
 endif
 
+# virtual-function-elimination reduces the size of binaries, but is still
+# experimental and has some possible miscompilation issues. We don't use it
+# by default but expose it via the `OPTIMISE=1 make` option.
+#
+# For details on virtual-function-elimination see: https://github.com/rust-lang/rust/pull/96285
+# See https://github.com/rust-lang/rust/issues/68262 for general tracking
+ifneq ($(OPTIMISE),)
```
Comment (hudson-ayers) [inline boards/Makefile.common:275]: Can we use `VFUNC_ELIM` instead of `OPTIMISE`? `OPTIMISE` is very generic

**Comment by hudson-ayers** [comment]
At a glance the netlify failure appears to be a real rustdoc error: https://app.netlify.com/sites/docs-tockosorg/deploys/62c376c0f1e20d0008fd820c

**Review by bradjc — DISMISSED** [review]

**Comment by alistair23** [comment]
Fixed the build failures

**Review by hudson-ayers — APPROVED** [review]

**Review by hudson-ayers — APPROVED** [review]

**Review by bradjc — APPROVED** [review]

**Pull request merged**